pkg/block/azure/adapter.go

```diff
@@ -347,8 +361,18 @@ func (a *Adapter) Copy(ctx context.Context, sourceObj, destinationObj block.Obje
 		return err
 	}
 
-	destClient := a.client.NewContainerClient(qualifiedDestinationKey.ContainerName).NewBlobClient(qualifiedDestinationKey.BlobURL)
-	sourceClient := a.client.NewContainerClient(qualifiedSourceKey.ContainerName).NewBlobClient(qualifiedSourceKey.BlobURL)
+	destContainerClient, err := a.clientCache.NewContainerClient(qualifiedDestinationKey.ContainerURL, qualifiedDestinationKey.ContainerName)
+	if err != nil {
+		return err
+	}
+	destClient := destContainerClient.NewBlobClient(qualifiedDestinationKey.BlobURL)
+
+	srcContainerClient, err := a.clientCache.NewContainerClient(qualifiedSourceKey.ContainerURL, qualifiedSourceKey.ContainerName)
+	if err != nil {
+		return err
+	}
+	sourceClient := srcContainerClient.NewBlobClient(qualifiedSourceKey.BlobURL)
+
```

Comment:
I think that creating a presigned URL is not needed anymore - we should be able to perform a copy using the credentials we have

Reply:
That's not always true. Using StorageAccessKey is still supported and this workaround is needed for that.